src/context/WalletProvider/KeepKey/components/Pin.tsx

```diff
@@ -135,7 +162,15 @@ export const KeepKeyPin = () => {
         <Text color='gray.500' translation={`walletProvider.keepKey.${translationType}.body`} />
         <SimpleGrid columns={3} spacing={6} my={6} maxWidth='250px' ml='auto' mr='auto'>
           {pinNumbers.map(number => (
-            <Button key={number} size='lg' p={8} onClick={() => handlePinPress(number)}>
+            <Button
+              key={number}
+              size='lg'
+              p={8}
+              onClick={() => {
+                handlePinPress(number)
+                setIsPinEmpty(!pinFieldRef.current?.value)
```

Comment:
Can we move this to `handlePinPress()`?

Additionally, `handlePinPress` should really be wrapped in `useCallback` (it's not code you wrote, just existing code that can be improved!).

Reply:
I was having issues with placing handlePinPress() elsewhere as the `isPinEmpty` ends up being out of sync with the actual entry.

Reply:
Added the useCallback